### PR TITLE
SI-8608 f interpolator emits plain old strings

### PR DESCRIPTION
**Pull request by som-snytt**
When invoking `format` is obviated by a lack of
formatting fields, then just degenerate to an
unenhanced string.

This means it doesn't cost anything to use
f"$$ordinary" in place of "$ordinary", which
may cause warnings under -Xlint.

Note that certain format literals, in particular
for line separator %n, are not actually literals and
can't be replaced at compile time.

Review by @xeno-by for macro expertise

### CODE REVIEW

File: test/files/run/t8608-no-format.scala [hunk 1]
```diff
@@ -0,0 +1,6 @@
+
+import Predef.{ augmentString => _ }
```
Comment (retronym) [inline test/files/run/t8608-no-format.scala]: Should add a comment to explain the clever testing mechanism.

Comment (som-snytt) [inline test/files/run/t8608-no-format.scala]: OK, but then I'll have to remember what I did.

Comment (lrytz) [inline test/files/run/t8608-no-format.scala]: ping @som-snytt 

the macro expands `f"hello"` to `scala.this.Predef.augmentString("hello").format()` (after type checking), but probably not fully qualified.

i'd be in favor of a test that depends less on the implementation of the macro; maybe a bytecode test that ensures `format` is not invoked?

Comment (som-snytt) [inline test/files/run/t8608-no-format.scala]: @lrytz You're right, I'll change the nature of the test.  For the record, I hadn't actually forgotten, that was just a whiff of humor.  Sometimes I do actually forget.

**Comment by som-snytt** [comment]
Make the test implementation-proof or at least -resilient.

**Comment by som-snytt** [comment]
I think a JavapTest will suffice.  If I didn't waste a quarter-hour on what triggered the text search on "format", namely, "for more information".  Ack!

**Comment by som-snytt** [comment]
Superseded by #3878 because of weird history effects.

**Pull request closed without merging**